sycl/test-e2e/bindless_images/vulkan_interop/depth_format.cpp

```diff
@@ -0,0 +1,390 @@
+      extMemOutHandle, syclexp::external_mem_handle_type::opaque_fd,
+      imgSizeBytes};
+#endif
+
+  // Extension: create interop memory handles.
+  syclexp::external_mem externalMemIn =
+      syclexp::import_external_memory(extMemInDesc, syclQueue);
+  syclexp::external_mem externalMemOut =
+      syclexp::import_external_memory(extMemOutDesc, syclQueue);
+
+  // Image descriptor - Vulkan depth texture mapped to single channel fp32
+  // image.
+  syclexp::image_descriptor imgDesc(globalSize, 1,
+                                    sycl::image_channel_type::fp32);
+
+  // Extension: map image memory handles.
+  syclexp::image_mem_handle imgMemIn =
+      syclexp::map_external_image_memory(externalMemIn, imgDesc, syclQueue);
+  syclexp::image_mem_handle imgMemOut =
+      syclexp::map_external_image_memory(externalMemOut, imgDesc, syclQueue);
+
+  // Extension: create the image and return the handle.
+  syclexp::unsampled_image_handle imgIn =
+      syclexp::create_image(imgMemIn, imgDesc, syclQueue);
+  syclexp::unsampled_image_handle imgOut =
+      syclexp::create_image(imgMemOut, imgDesc, syclQueue);
+
+  try {
+    syclQueue.submit([&](sycl::handler &cgh) {
+      cgh.parallel_for<class TestDepthTextureFetch>(
+          sycl::nd_range<2>{globalSize, localSize}, [=](sycl::nd_item<2> it) {
+            size_t dim0 = it.get_global_id(0);
+            size_t dim1 = it.get_global_id(1);
+
+            float depth =
+                syclexp::fetch_image<float>(imgIn, sycl::int2(dim0, dim1));
+
+            syclexp::write_image<float>(imgOut, sycl::int2(dim0, dim1), depth);
+          });
+    });
+
+    // Wait for kernel completion before destroying external objects.
+    syclQueue.wait_and_throw();
+
+    // Cleanup.
+    syclexp::destroy_image_handle(imgIn, syclQueue);
+    syclexp::destroy_image_handle(imgOut, syclQueue);
+    syclexp::free_image_mem(imgMemIn, syclexp::image_type::standard, syclQueue);
+    syclexp::free_image_mem(imgMemOut, syclexp::image_type::standard,
+                            syclQueue);
+    syclexp::release_external_memory(externalMemIn, syclQueue);
+    syclexp::release_external_memory(externalMemOut, syclQueue);
+  } catch (sycl::exception e) {
+    std::cerr << "\tKernel submission failed! " << e.what() << std::endl;
+    exit(-1);
+  } catch (...) {
+    std::cerr << "\tKernel submission failed!" << std::endl;
+    exit(-1);
+  }
+}
+
+bool runTest(const sycl::device &syclDevice, sycl::range<2> dims,
+             sycl::range<2> localSize) {
+  const uint32_t imgWidth = static_cast<uint32_t>(dims[0]);
+  const uint32_t imgHeight = static_cast<uint32_t>(dims[1]);
+
+  const VkImageType imgType = VK_IMAGE_TYPE_2D;
+  const VkFormat imgInFormat = VK_FORMAT_D32_SFLOAT;
+  const VkFormat imgOutFormat = VK_FORMAT_D32_SFLOAT;
+
+  const size_t imgSizeElems = imgWidth * imgHeight;
+  const size_t imgSizeBytes = imgSizeElems * sizeof(float);
+
+  const VkExtent3D imgExtent = {imgWidth, imgHeight, 1};
```

Comment:
Why is `VkExtent3D` used here for what is eventually a 2D image instead of `VkExtent2D`?

Reply:
[`VkImageCreateInfo`](https://registry.khronos.org/vulkan/specs/latest/man/html/VkImageCreateInfo.html) takes a `VkExtent3D`.

Reply:
Interesting, so it does. Fair enough.